### PR TITLE
Add Swift adoption docs and tests

### PR DESCRIPTION
**Pull request by cschwede**
Adds documentation and tests to adopt an existing Swift object storage workload. This does not yet include migrating data itself.

### CODE REVIEW

File: docs_user/modules/openstack-swift_adoption.adoc [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+'
+----
+
+== Post-checks
+
+=== Inspect the resulting swift service pods
+
+[,bash]
+----
+oc get pods -l app.kubernetes.io/name=SwiftProxy
```
Comment (cschwede) [inline docs_user/modules/openstack-swift_adoption.adoc:109]: This needs to be changed once https://github.com/openstack-k8s-operators/swift-operator/pull/126 is merged and the openstack-operator bumped the swift-operator version. Should be "oc get pods -l component=swift-proxy" then

File: tests/roles/swift_adoption/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                    type: LoadBalancer
+            networkAttachments:
+            - storage
+    '
+
+- name: Wait for Swift to start up
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc get pod -l app.kubernetes.io/name=SwiftProxy -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
```
Comment (cschwede) [inline tests/roles/swift_adoption/tasks/main.yaml:75]: This needs to be changed once https://github.com/openstack-k8s-operators/swift-operator/pull/126 is merged and the openstack-operator bumped the swift-operator version. Should be "oc get pods -l component=swift-proxy" then

File: docs_user/modules/openstack-swift_adoption.adoc (hunk 1 above)
Comment (cschwede) [reply docs_user/modules/openstack-swift_adoption.adoc]: Done

File: tests/roles/swift_adoption/tasks/main.yaml (hunk 2 above)
Comment (cschwede) [reply tests/roles/swift_adoption/tasks/main.yaml]: Done

**Comment by cschwede** [comment]
recheck

**Review by jistr — APPROVED** [review]
Looks good, thanks! :+1:

**Review by fao89 — APPROVED** [review]

**Pull request merged**